src/IgnoreList.cxx

```diff
@@ -0,0 +1,40 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+// Copyright The Music Player Daemon Project
+
+#include "IgnoreList.hxx"
+
+#include <cassert>
+#include <ranges>
+#include <algorithm>
+
+bool IgnoreListEntry::matches_record(const Record& record) const {
+	// The below logic would always return true if the entry is empty.
+	// This condition should never be true, as we don't push empty entries.
+	assert(this->artist || this->album || this->title);
```

Comment:
Please don't use `this->` everywhere, it's redundant noise.

Reply:
Will fix.